tutorials/sphinx_tuto/tensorclass_imagenet.py

```diff
@@ -153,59 +250,132 @@ def __len__(self):
+        t0 = time.time()
+    if i >= 3:
+        total += image.shape[0]
     image, target = image.to(device), target.to(device)
-print(f"One iteration over dataloader done! Time: {time.time() - t0:4.4f}s")
-
+t = time.time() - t0
+print(f"One iteration over dataloader done! Rate: {total/t:4.4f} fps, time: {t: 4.4f}s")
 
 ##############################################################################
 # Our tensorclass-based dataloader instead loads data from the
 # memory-mapped tensor in batches. We then apply the batched random
 # transformations to the batched images.
 
-t0 = time.time()
-for batch in train_dataloader_tc:
-    image, target = (
-        batch_train_transform(batch.images.contiguous()),
-        batch.targets.contiguous(),
-    )
-print(f"One iteration over tensorclass dataloader done! Time: {time.time() - t0:4.4f}s")
+total = 0
+for i, batch in enumerate(train_dataloader_tc):
+    if i == 3:
+        t0 = time.time()
+    if i >= 3:
+        total += batch.numel()
+    image, target = batch.images, batch.targets
+t = time.time() - t0
+print(
+    f"One iteration over tensorclass dataloader done! Rate: {total/t:4.4f} fps, time: {t: 4.4f}s"
+)
 
 ##############################################################################
 # In the case of the validation set, we see an even bigger performance
 # improvement, because there are no random transformations, so we can save
 # the fully transformed data in the memory-mapped tensor, eliminating the
 # need for additional transformations as we load from disk.
 
-t0 = time.time()
-for image, target in val_dataloader:
+total = 0
+for i, (image, target) in enumerate(val_dataloader):
+    if i == 3:
+        t0 = time.time()
+    if i >= 3:
+        total += image.shape[0]
     image, target = image.to(device), target.to(device)
-print(f"One iteration over val data. Time: {time.time() - t0:4.4f}s")
+t = time.time() - t0
+print(f"One iteration over val data done! Rate: {total/t:4.4f} fps, time: {t: 4.4f}s")
+
+total = 0
+for i, batch in enumerate(val_dataloader_tc):
+    if i == 3:
+        t0 = time.time()
+    if i >= 3:
+        total += batch.shape[0]
+    image, target = batch.images.contiguous().to(device), batch.targets.contiguous().to(
+        device
+    )
+t = time.time() - t0
+print(
+    f"One iteration over tensorclass val data done! Rate: {total/t:4.4f} fps, time: {t: 4.4f}s"
+)
 
-t0 = time.time()
-for batch in val_dataloader_tc:
-    image, target = batch.images.contiguous(), batch.targets.contiguous()
-print(f"One iteration over tensorclass val data. Time: {time.time() - t0:4.4f}s")
+##############################################################################
+# Results from ImageNet
+# ---------------------
+#
+# Repeating the above on full-size ImageNet data, we get the following results
+#
+#  | One iteration over dataloader done! Rate: 2984.8455 fps, time:  428.7951s
+#  | One iteration over tensorclass dataloader done! Rate: 6457.9865 fps, time:  198.1867s
```

Comment:
I have new results using a mp collector for the tensordict version:


```
One iteration over dataloader done! Rate: 2969.6382 fps, time:  430.9909s
One iteration over tensorclass dataloader done! Rate: 27273.2961 fps, time:  46.9282s
One iteration over val data done! Rate: 2367.5322 fps, time:  20.5784s
One iteration over tensorclass val data done! Rate: 19623.9993 fps, time:  2.4827s
```

We are 10x as fast!